File: Blind 75/Invert Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        // switch the left and right children nodes and 
        // recursively call left then right
        if (root == null) return null;
        TreeNode temp = root.left;
        root.left = root.right;
        root.right = temp;
        // more depth first since we go through all the left children 
        // nodes then the right
        invertTree(root.left);
        invertTree(root.right);
        return root;
    }
}

File: Python Blind 75/Tree/Invert Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        # Time: O(n), Space: O(n)
        if not root: return None
        root.left, root.right = root.right, root.left # No need to make a temp var to switch
        self.invertTree(root.left)
        self.invertTree(root.right)
        return root